File: src/main/java/aurocosh/divinefavor/common/entity/minions/behaviour/base/MinionBehaviour.java
package aurocosh.divinefavor.common.entity.minions.behaviour.base;

import aurocosh.divinefavor.common.entity.minions.base.IMinion;
import net.minecraft.entity.EntityLiving;
import net.minecraft.entity.ai.EntityAITasks;

public abstract class MinionBehaviour <T extends EntityLiving & IMinion> {
    public abstract void apply(T minion, EntityAITasks tasks, EntityAITasks targetTasks);
}

File: src/main/kotlin/aurocosh/divinefavor/common/entity/minions/behaviour/base/MinionBehaviour.kt
package aurocosh.divinefavor.common.entity.minions.behaviour.base

import aurocosh.divinefavor.common.entity.minions.base.IMinion
import net.minecraft.entity.EntityLiving
import net.minecraft.entity.ai.EntityAITasks

abstract class MinionBehaviour<T> where T : IMinion, T : EntityLiving {
    abstract fun apply(minion: T, tasks: EntityAITasks, targetTasks: EntityAITasks)
}
